// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.jpeg";
option java_outer_classname = "JpegProto";
option objc_class_prefix = "JP";

package jpeg;

// The greeting service definition.
service Greeter {
  rpc SendImg (ImgRequest) returns (ImgReply) {}
  rpc SendDirectoryContents (DirectoryContentsRequest) returns (DirectoryContentsReply) {}
}

message ImgRequest {
  string name = 1;
}

message ImgReply {
  int32 size = 1;  
  bytes img = 2;
  string name = 3;
  string date = 4;
}

message DirectoryContentsRequest{
  string id = 1;
  string pw = 2;
}
message DirectoryContentsReply{
  repeated string directory = 1; // 배열. 디렉토리 안에 있는 
  int32 count = 2;
}
